### PR TITLE
Operation 'get pacts green'

### PR DESCRIPTION
**Pull request by tomsmyers**
## What does this pull request do?

major refactoring of the pact contract test code - see the commit messages for more details. 

there is currently only _one_ functional change in the application code, this was picked up by one of the failing tests, so that's good news.

before: 
Screenshot 2021-04-13 at 00 09 08

## What is the intent behind these changes?

get the pact tests green

### CODE REVIEW

**Review by mahalrmoj — APPROVED** [review]
Pact is green - hooooooray

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/jpa/repository/DesiredOutcomeRepository.kt [hunk 1]
```diff
@@ -4,4 +4,6 @@ import org.springframework.data.jpa.repository.JpaRepository
 import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.DesiredOutcome
 import java.util.UUID
 
-interface DesiredOutcomeRepository : JpaRepository<DesiredOutcome, UUID>
+interface DesiredOutcomeRepository : JpaRepository<DesiredOutcome, UUID> {
+  fun findByServiceCategoryId(serviceCategoryId: UUID): List<DesiredOutcome>
```
Comment (sldblog) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/jpa/repository/DesiredOutcomeRepository.kt:8]: Would we also use this in a production-path scenario? Just thinking if this is a hint towards something deeper if we find it useful for contract testing

**Comment by sldblog** [comment]
> Pact is green - hooooooray

Let's make it required? 😁

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/jpa/repository/DesiredOutcomeRepository.kt (hunk 1 above)
Comment (tomsmyers) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/jpa/repository/DesiredOutcomeRepository.kt:8]: i wrote this with somewhat bleary eyes - i think the correct way to get the desired outcomes is just `serviceCategoryRepository.findById({id}).desiredOutcomes`

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/integration/pact/ReferralContracts.kt [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    val intervention = setupAssistant.createIntervention()
+    val desiredOutcomes = setupAssistant.desiredOutcomesForServiceCategory(intervention.dynamicFrameworkContract.serviceCategory.id)
+    val referral = setupAssistant.createDraftReferral(id = UUID.fromString("037cc90b-beaa-4a32-9ab7-7f79136e1d27"))
+    setupAssistant.fillReferralFields(referral, desiredOutcomes = desiredOutcomes)
+  }
+
+  @State("There is an existing draft referral with ID of 1219a064-709b-4b6c-a11e-10b8cb3966f6, and it has had a service user selected")
+  fun `create a new draft referral with the service user data expected`() {
+    val referral = setupAssistant.createDraftReferral(id = UUID.fromString("1219a064-709b-4b6c-a11e-10b8cb3966f6"))
+    setupAssistant.fillReferralFields(referral)
```
Comment (sldblog) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/integration/pact/ReferralContracts.kt:60]: Should we rethink how we describe our `@State`s?

I.e. it sounds like "an empty draft referral with ID ..." and "an existing ready-to-send draft referral with ID ..." encapsulates most of these states

**Pull request merged**